### DISCUSSION TITLE
Unit Testing

### DISCUSSION THREAD

**Discussion by gfrances** [rleap-project/dlplan#7] [Ideas]
Hi guys,

I've added some code to integrate the project with google test, plus one quick test to see how this could look like.
It's all in the `gtest` branch. The code is fairly simple, and makes use of some interesting `FetchContent` CMake module that I didn't know existed until a while ago, which really simplifies things. I've been able to run this from the console and integrate this with CLion with no trouble at all. The only thing I'm concerned with is that `FetchContent` [seems to be CMake 3.11+-only](https://cliutils.gitlab.io/modern-cmake/chapters/projects/fetch.html). This version is around 2.5 years old, and hence not included in e.g. Ubuntu 18.04LTS.

An alternative to using this would be to include the GTest source tree within our project tree, as a git submodule (maybe better), or manually as a copy-paste.

What do you guys think?

**Comment by jendrikseipp** [comment]
Nice! I think it's fine to require CMake 3.11. People on older Ubuntu versions can use the package archive: https://apt.kitware.com/

**Comment by drexlerd** [comment]
I will change back to 3.11 from 3.16 then if this works already. I like the gtest framework a lot because it has always been easy to use for me under Linux. I remember though that I had some troubles under Windows. But I don't know exactly what went wrong there.
Either way is fine for me.
